### PR TITLE
Di 20 monster hca dev env

### PR DESCRIPTION
**Pull request by bahill**
## Why

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DI-20)
Adding Docker Compose Dev env to speed up and simplify dev env setup

## Checklist
- [ x] Documentation has been updated as needed.

### CODE REVIEW

**Comment by bahill** [comment]
OK - time to fight with the linter

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,127 @@
+## Builds the hca-ingest dev env image
+
+# This is the Dagster user code deployment image
+# Dockerfile is located in orchestration/Dockerfile
+# FROM us.gcr.io/broad-dsp-gcr-public/monster-hca-dagster:latest
+FROM us.gcr.io/broad-dsp-gcr-public/monster-hca-dagster:WIP_bhill
```
Comment (aherbst-broad) [inline Dockerfile:6]: Will this eventually point at non-WIP tag?

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+services:
```
Comment (aherbst-broad) [inline docker-compose.yaml:1]: I'm curious why docker compose is needed if the only service being run is a single container.

**Review by aherbst-broad — COMMENTED** [review]
I can't view the relevant ticket, can you provide a bit more context about why this is necessary vs running in a venv?

File: docker-compose.yaml (hunk 2 above)
Comment (bahill) [reply docker-compose.yaml:1]: Using Docker compose is allowing me to work in my local git clone but use the cached dev env.
https://github.com/ArjanCodes/2022-docker

I could also have used --reload in a run command, but I like the syncing this method provides, and the configuration of the run via a file vs a long command line.

File: Dockerfile (hunk 1 above)
Comment (bahill) [reply Dockerfile]: Ok that's all set now. Thank you for catching that!

**Review by aherbst-broad — APPROVED** [review]
lgtm, thanks for the clarification.

**Pull request merged**